src/main/java/htsjdk/samtools/util/Histogram.java

```diff
@@ -42,40 +35,46 @@
  *
  * @author Tim Fennell
  */
-public class Histogram<K extends Comparable> extends TreeMap<K, Bin> {
+public final class Histogram<K extends Comparable<K>> implements Serializable {
```

Comment:
Recommend that we not cause any more breakage here than absolutely necessary to fix the original issue(s). We'll be using Kryo anyway to serialize this in GATK, which does not require strict Java serializability -- as long as we fix the non-static inner class and the issue related to inheriting from TreeMap we should be fine.